cli/src/main/java/com/devonfw/tools/ide/context/AbstractIdeContext.java

```diff
@@ -824,7 +824,7 @@ private boolean applyAndRun(CliArguments arguments, Commandlet cmd) {
 
     boolean matches = apply(arguments, cmd, null);
     if (matches) {
-      matches = cmd.validate();
+      matches = cmd.validate().isValid();
```

Comment:
IMHO this needs to go even further.
So also here we need to return `ValidationResult`.
Tricky thing is if `matches` is currently `false` but not from validation, we would even need a reason as `validationError`.
So most likely also `apply` needs to return `ValidationResult`.
Then `run(CliArguments)` can be changed to handle the `ValidationResult`
So instead of `boolean matches` declare the `ValidationResult` and keep the one from the `firstCandidate`.
Now instead at the
```
step.error("Invalid arguments: {}", current.getArgs());
```
We can now handle that `ValidationResult` if not valid and get the error message.
The error message could be appended to the error log message or IMHO even better, we could throw a CliException here from the ValidationResult. If we want to make that reusable, we can even derive a class `CliValidationException` that can be created from `ValidationResult`, `Commandlet`, and `args` to create a reasonable exception.
This makes it better for JUnit tests since we can catch such exception to assert validation error behavior instead of checking status codes and searching for log messages what is less stable.